### DISCUSSION TITLE
[FR]: Different format for display and store in Date and Time Field.

### DISCUSSION THREAD

**Discussion by uroybd** [RafaelGB/obsidian-db-folder#698] [Q&A]
### Contact Details

roy@utsob.me

### Present your request

Often we choose to store the Date or Time in accurate ISO formats like `2022-12-17T20:06:05+06:00`. In the database table, they are not easily readable. Now we have two options to deal with it with tread-offs:

1. Leave it as is. It will be less readable but won't be messed up on edit.
2. Use a formula field and show them formatted as you like (my current choice). But you lose editability.

I propose optional field(s) to determine how the Date or Time will be shown in the table. One can choose the date or time from the widget and it will save using the current date and time formatting settings and show the fields formatted with these new settings if available.

### For which platform do you request this request??

Desktop

**Comment by RafaelGB** [comment]
You can already choose your favourite format to display here

<img width="775" alt="Screenshot 2022-12-26 at 12 18 11" src="https://user-images.githubusercontent.com/11924043/209543128-55a6a842-4bd5-416d-96b2-4908df602cae.png">

Since 3.1.0 its true we save the date on file as ISO, but it will be resolved with the next one https://github.com/RafaelGB/obsidian-db-folder/issues/683

**Reply by uroybd** [reply]
> You can already choose your favourite format to display here
> 
> <img alt="Screenshot 2022-12-26 at 12 18 11" width="775" src="https://user-images.githubusercontent.com/11924043/209543128-55a6a842-4bd5-416d-96b2-4908df602cae.png">
> 
> Since 3.1.0 its true we save the date on file as ISO, but it will be resolved with the next one #683

Fixing this is nice (and I can see that in the last update) but that was not what I was asking.

I want to format the set and view it differently. For Example, on edit, I want to set the DateTime like this: `2022-12-27T12:30:00+06:00`. However, in columns, I want to see it as `Dec 27, 2022 12:30PM`. This was the behaviour before the fix. However, that is confusing to many. Almost always, ISO formats are preferable in frontmatter. All the static site generator expects it that way. This is also the most unambiguous of all time formats.

**Reply by RafaelGB** [reply]
my opinion is that separating the behavior between writing and parsing requires increasing the combinatorics of parsing and can be misleading
